pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -101,6 +101,8 @@ type reconcilerBase struct {
 
 	// syncKind is the kind of the sync object: RootSync or RepoSync.
 	syncKind string
+
+	reconcilerFinalizerHandler
```

Comment:
huh. i didn't know it was possible to have multiple "parent" structs with inherited methods. TIL.

Might make more sense to move this to the top with loggingController though.

Alternatively, do the strategy methods actually need to be available on the reconcilerBase, or would it be better to just keep them private so the caller can't run them?

Reply:
done, keep the function private to reconcilerBase.